internal/service/identitystore/groups_data_source_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package identitystore_test
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"testing"
+
+	sdkacctest "github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	tfidentitystore "github.com/hashicorp/terraform-provider-aws/internal/service/identitystore"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func TestAccIdentityStoreGroupsDataSource_basic(t *testing.T) {
+	ctx := acctest.Context(t)
+
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix) // Group Name
+	dataSourceName := "data.aws_identitystore_groups.test"
+	groupResourceName := "aws_identitystore_group.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckSSOAdminInstances(ctx, t)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, names.IdentityStoreServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		// All destroy checks are already covered in sibling tests
+		CheckDestroy: acctest.CheckDestroyNoop,
```

Comment:
I couldn't find the implementing of CheckDestroy, is it covered somewhere?

Reply:
Will look into it!

Reply:
I've used the CheckDestroy now which is used in the `group_data_source` tests, seems to cover the functionality needed